joblib/test/test_memory.py

```diff
@@ -410,33 +387,38 @@ def test_persistence(tmpdir):
     assert memory.cachedir == memory2.cachedir
 
     # Smoke test that pickling a memory with cachedir=None works
-    memory = Memory(cachedir=None, verbose=0)
-    pickle.loads(pickle.dumps(memory))
+    pickle.loads(pickle.dumps(no_memoize_memory))
     g = memory.cache(f)
     gp = pickle.loads(pickle.dumps(g))
     gp(1)
 
 
-def test_call_and_shelve(tmpdir):
+@fixture(scope='function', params=range(4))
```

Comment:
When I said `parametrize` I meant rather a fixture with a `params` parameter, but you get the point ...